src/runtime/recipe/handle-connection.ts

```diff
@@ -17,23 +17,23 @@ import {compareArrays, compareComparables, compareStrings} from './comparable.js
 export type Direction = 'in' | 'out' | 'inout' | 'host';
 
 export class HandleConnection {
-  _recipe: Recipe;
-  _name: string;
-  _tags: string[] = [];
+  private readonly _recipe: Recipe;
+  private readonly _name: string;
+  private _tags: string[] = [];
   private resolvedType: Type | undefined = undefined;
-  _direction: Direction | undefined = undefined;
-  _particle: Particle;
+  private _direction: Direction | undefined = undefined;
+  private _particle: Particle;
   _handle: Handle | undefined = undefined;
 
-  constructor(name, particle) {
+  constructor(name: string, particle: Particle) {
     assert(particle);
     assert(particle.recipe);
     this._recipe = particle.recipe;
     this._name = name;
     this._particle = particle;
   }
 
-  _clone(particle, cloneMap) {
+  _clone(particle: Particle, cloneMap) {
```

Comment:
If `_clone` in the above suggestion gets a type, then so should this fn.

Reply:
`cloneMap` seems like it might be tricky to type.
Is it an `object`? If it's commonly occurring, maybe we can alias `object` (or it's actual type).

Reply:
agreed, more room for improvement.  For  a comprehensive list you can run `./tools/sigh health --types`